### DISCUSSION TITLE
[Gradle] UseHttpsForRepositories recipe doesn't work because it can't resolve dependencies via http

### DISCUSSION THREAD

**Discussion by marcorichetta** [openrewrite/rewrite#3637] [Q&A] [answered]
Hi!
I'm trying to run the [UseHttpsForRepositories](https://docs.openrewrite.org/recipes/gradle/security/usehttpsforrepositories) but it fails with the following error. [Gradle docs](https://docs.gradle.org/7.3/dsl/org.gradle.api.artifacts.repositories.UrlArtifactRepository.html#org.gradle.api.artifacts.repositories.UrlArtifactRepository:allowInsecureProtocol)

```bash
❯ gradle --init-script init.gradle rewriteRun

FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring root project 'gradle'.
> Could not resolve all dependencies for configuration ':classpath'.
   > Using insecure protocols with repositories, without explicit opt-in, is unsupported. Switch Maven repository 'maven(http://maven.artifacts.mydomain.com/repository/all)' to redirect to a secure protocol (like HTTPS) or allow insecure protocols. 
	See https://docs.gradle.org/7.3/dsl/org.gradle.api.artifacts.repositories.UrlArtifactRepository.html#org.gradle.api.artifacts.repositories.UrlArtifactRepository:allowInsecureProtocol for more details.
```

I'd expect to be able to execute this recipe on projects that have http repositories. If not, what's the purpose of it?

#### build.gradle
```groovy
buildscript {
    repositories {
        maven { url "http://maven.artifacts.mydomain.com/repository/all" }
    }
}
```

I'm executing it like an initscript pretty much the same as the one in the docs.

Thanks!

**Comment by marcorichetta** [comment, marked as answer]
I was running it with Gradle 6.3. I forgot I can use the gradle wrapper of the project, which is v6.8 in this case.
After that I could successfully run the recipe

**Reply by timtebeek** [reply]
Glad to hear you were able to run the recipe to replace the insecure repository url. You might want to follow up with this [recipe to upgrade the Gradle wrapper](https://docs.openrewrite.org/recipes/gradle/updategradlewrapper), if you haven't already.

For a bit of context: we don't often see folks run these recipes on a single project, instead opting to run these recipes across multiple projects simultaneously using [the Moderne CLI](https://docs.moderne.io/moderne-cli/cli-intro), or the Moderne platform at https://app.moderne.io . That might explain why we have these recipes, if there might be some difficulties running these recipes depending on your Gradle version.

For any future users: running might also have worked when passing in `allowInsecureProtocol=true` to Gradle, only for the duration of the recipe run. Hope that helps!
